### ISSUE TITLE
Add an option for dropping into a debugger only after an error/failure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6143]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)| @Julian reported|
|-|-|
|Trac ID|trac#6143|
|Type|enhancement|
|Created|2012-10-17 00:24:19Z|

See nosetests --pdb and --pdb-failures and py.test --pdb which both do it this way, rather than needing to step through test runner code.

I personally don't really use them independently of one another (like nose has it) so IMHO just a single option for dropping in either case into a debugger would be fine.

Since --debug is taken, I can't think of a great name immediately. --debug-non-successes is kinda long.

<details><summary>Searchable metadata</summary>

```
trac-id__6143 6143
type__enhancement enhancement
reporter__Julian Julian
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__trial trial
keywords__easy easy
time__1350433459000000 1350433459000000
changetime__1366649638000000 1366649638000000
version__None None
owner__mayank mayank
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_2"></a>|@Julian commented|
|-|-|

[#2230](https://github.com/twisted/twisted/issues/2230) was a duplicate of this. When this lands, we should make sure the desired behavior there re: `--until-failure` works properly.

**Comment by twisted-trac** [comment]
|<img alt="mayank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|mayank set owner to mayank<br>mayank set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set status to `closed`|
|-|-|

Since [#2330](https://github.com/twisted/twisted/issues/2330) is the much older ticket here, I'm closing this one as a duplicate instead.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Replying to [glyph](#note_5):
> Since [#2330](https://github.com/twisted/twisted/issues/2330) is the much older ticket here, I'm closing this one as a duplicate instead.

Ahem. [#2230](https://github.com/twisted/twisted/issues/2230).
